components/Healthlog/SubtypeDetail/index.tsx

```diff
@@ -32,25 +47,36 @@ const SubtypeDetail: React.FC<SubtypeDetailProps> = ({ visibleSubtype, register
         </div>
       )}
       {["FEED", "HEALTH", "TREAT", "GROOMING"].includes(visibleSubtype) && (
-        <div className={styles.inputWrapper}>
-          <label>세부 사항</label>
-          <select className={styles.selectBox} onChange={handleSelectChange} value={selectedOption}>
-            {subtypeOptions[visibleSubtype as keyof typeof subtypeOptions].map((option: string, index: number) => (
-              <option key={index} value={option}>
-                {option}
-              </option>
-            ))}
-          </select>
-          {selectedOption === "직접 입력" && <input className={styles.inputBox} {...register("subtype")} />}
+        <div className={styles.selectWrapper} ref={dropdownRef}>
+          <button className={styles.selectBox} onClick={() => setDropdownOpen(!dropdownOpen)}>
+            {selectedOption || "세부사항 선택"}
+          </button>
+          {dropdownOpen && (
+            <ul className={styles.optionsList}>
+              {subtypeOptions[visibleSubtype as keyof typeof subtypeOptions].map((option: string, index: number) => (
```

Comment:
바로 subtypeOptions[visibleSubtype]을 하면 안 되고 `as keyof typeof`로 타고 들어가줘야 하나요?

 데이터를 조건마다 다르게 다뤄야 하는게 어려워 보이는데 고생이 많으세요 소희님💓

Reply:
처음에 그렇게 했더니 아래 타입 오류가 뜨더라구요. 그래서 저렇게 해줬어요 ㅠ 뭔가 지저분해서 맘에 들진 않는데 대안이 있을지 모르겠네요 ㅠ 멘토님께 여쭤봐야겠어요.

Element implicitly has an 'any' type because expression of type '"FEED" | "HEALTH" | "TREAT" | "GROOMING" | "CUSTOM" | "WALK"' can't be used to index type '{ FEED: string[]; HEALTH: string[]; TREAT: string[]; GROOMING: string[]; }'.
  Property 'CUSTOM' does not exist on type '{ FEED: string[]; HEALTH: string[]; TREAT: string[]; GROOMING: string[]; }'.